mailbox/caching/src/main/java/org/apache/james/mailbox/caching/CacheInvalidatingMailboxListener.java

```diff
@@ -41,10 +41,14 @@ public void register(EventBus eventBus) {
         eventBus.register(this);
     }
 
+    @Override
+    public boolean isHandling(Event event) {
+        return event instanceof MailboxEvent;
+    }
 
     @Override
     public void event(Event event) {
-        if (event instanceof MailboxEvent) {
+        if (isHandling(event)) {
```

Comment:
does it make sense? The caller should have make that check already

Reply:
The caller do not always call the isHandling method. It's used has a pre-filter to avoid counting a non handled event in the metrics. But not all callers have this concern.

Reply:
Are you sure? I do think the caller is always EventBus and we could very well enforce a call to `isHandling` by writing the related contract test, don't you think?